### PR TITLE
include locationRef in details merger

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

Adds details merging handling to locationRef.

## Does this PR need a docs update or release note?

- [x] :clock1: Yes, but in a later PR

## Type of change

- [x] :sunflower: Feature

## Issue(s)

* #2423

## Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Review by ashmrtn — CHANGES_REQUESTED** [review]
need to fix at least the call to `FolderEntriesForPath` in backup merge details

File: src/internal/operations/backup.go [hunk 1]
```diff
@@ -552,28 +552,39 @@ func mergeDetails(
...
 			// TODO(ashmrtn): This may need updated if we start using this merge
 			// strategry for items that were cached in kopia.
-			itemUpdated := newPath.String() != rr.String()
+			var (
+				itemUpdated = newPath.String() != rr.String()
+				newLocStr   string
+			)
+
+			if newLoc != nil {
+				newLocStr = newLoc.Folder()
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/operations/backup.go:579]: not necessary.  We'll have to handle that transition when we merge the branches together anyway.  (this is turning into more of a merge tree than a merge train).

[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/operations/backup.go:579]: not necessary.  We'll have to handle that transition when we merge the branches together anyway.  (this is turning into more of a merge tree than a merge train).

File: src/internal/operations/backup.go [hunk 2]
```diff
@@ -552,28 +552,39 @@ func mergeDetails(
...
+				itemUpdated = itemUpdated || newLocStr != entry.LocationRef
+			}
 
 			deets.Add(
 				newPath.String(),
 				newPath.ShortRef(),
 				newPath.ToBuilder().Dir().ShortRef(),
-				"", // TODO Location Ref,
+				newLocStr,
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/operations/backup.go:587]: Good catch, thank you.

File: src/internal/operations/backup_test.go [hunk 3]
```diff
@@ -273,16 +273,22 @@ func makePath(t *testing.T, elements []string, isItem bool) path.Path {
...
+	if l != nil {
+		lr = l.PopFront().PopFront().PopFront().PopFront().Dir().String()
+	}
+
 	res := &details.DetailsEntry{
 		RepoRef:     p.String(),
 		ShortRef:    p.ShortRef(),
 		ParentRef:   p.ToBuilder().Dir().ShortRef(),
-		LocationRef: p.PopFront().PopFront().PopFront().PopFront().Dir().String(),
+		LocationRef: lr,
```
Comment (ashmrtn) [inline src/internal/operations/backup_test.go:291]: ahhh now that this is a param I'm super tempted to ask for them not to be the same values in the tests so we know the code isn't conflating the two (or ignoring one of them). This kind of is along the lines of "what assumptions does my test make about the ways the code could possibly be broken?" But anyway, not a blocker for this to merge

**Review by ashmrtn — APPROVED** [review]

File: src/internal/operations/backup_test.go (hunk 3 above)
Comment (ryanfkeepers) [reply src/internal/operations/backup_test.go:291]: eh, might as well

Comment (ryanfkeepers) [reply src/internal/operations/backup_test.go:291]: totally worth it: caught a bug

**Pull request merged**